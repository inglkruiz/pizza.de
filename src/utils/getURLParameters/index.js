// @flow
/**
 * Transforms a URL query string to an object
 * @param {String} queryString URL query parameters
 * @returns {Object}
 */
export default function getURLParameters(
  queryString: string = ''
): { [string]: string } {
  const paramsObject = {}
  queryString = queryString.replace(/^\?/, '')

  if (!queryString) return paramsObject

  const dataURL = queryString.split('&')
  let data, keyValue, key

  for (data of dataURL) {
    keyValue = data.split('=')
    key = keyValue[0]

    if (key) {
      paramsObject[key.toLowerCase()] = decodeURIComponent(keyValue[1])
    }
  }

  return paramsObject
}
