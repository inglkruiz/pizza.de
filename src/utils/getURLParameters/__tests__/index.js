import getURLParameters from '../index'

test('Returns an empty object when there is not any query parameter', () => {
  expect(getURLParameters()).toEqual({})
  expect(getURLParameters('?')).toEqual({})
})

test('Returns an object with attributes as the query parameters passed', () => {
  const queryObj = getURLParameters('param1=value1&param2=value2&param3=value3')
  expect(queryObj).toMatchSnapshot()
})
