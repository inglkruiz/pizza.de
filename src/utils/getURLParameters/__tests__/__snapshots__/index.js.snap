// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Returns an object with attributes as the query parameters passed 1`] = `
Object {
  "param1": "value1",
  "param2": "value2",
  "param3": "value3",
}
`;
